### PR TITLE
[UNR-5021] Use GDK heartbeat configurations for worker timeout

### PR DESCRIPTION
**Pull request by razvan-at-improbable**
#### Description
Changed the way worker heartbeat parameters are initialized to use spatial gdk settings.

#### Release note
Added use of the heartbeat settings found in the editor for worker heartbeat parameters.

#### Tests
STRONGLY SUGGESTED: How can this be verified by QA?
Set "heartbeat timeout with editor" to a low value, put a breakpoint in any tick function (Potentially in Actor.cpp's ExecuteTick), wait long enough, after which continue and make sure the workers have indeed disconnected via the Inspector Worker view.
#### Documentation
Not needed.

#### Primary reviewers
@m-samiec

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp [hunk 1]
```diff
@@ -101,17 +101,16 @@ struct ConfigureConnection
...
 		TimeoutSeconds = Settings->HeartbeatTimeoutSeconds;
 #endif
-		Worker_HeartbeatParameters WorkerHeartbeatParams = { static_cast<uint64_t>(Settings->HeartbeatIntervalSeconds
-																							   * SECONDS_TO_MILLIS),
-																	static_cast<uint64_t>(TimeoutSeconds * SECONDS_TO_MILLIS) };
-		Params.network.tcp.downstream_heartbeat = &WorkerHeartbeatParams;
-		Params.network.tcp.upstream_heartbeat = &WorkerHeartbeatParams;
-		Params.network.kcp.downstream_heartbeat = &WorkerHeartbeatParams;
-		Params.network.kcp.upstream_heartbeat = &WorkerHeartbeatParams;
+		Worker_HeartbeatParameters HeartbeatParams = { static_cast<uint64_t>(Settings->HeartbeatIntervalSeconds * 1000),
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp:108]: To my first comment, this object is on the stack and will be destroyed when the function returns, so needs to remain a member variable.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp [hunk 2]
```diff
@@ -101,17 +101,16 @@ struct ConfigureConnection
 		float TimeoutSeconds;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp:101]:
```suggestion
```

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp [hunk 3]
```diff
@@ -101,17 +101,16 @@ struct ConfigureConnection
...
 #if WITH_EDITOR
 		TimeoutSeconds = Settings->HeartbeatTimeoutWithEditorSeconds;
-		
+
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp:104]:
```suggestion
```

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp [hunk 4]
```diff
@@ -101,17 +101,16 @@ struct ConfigureConnection
 		float TimeoutSeconds;
 #if WITH_EDITOR
 		TimeoutSeconds = Settings->HeartbeatTimeoutWithEditorSeconds;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp:103]:
```suggestion
		float TimeoutSeconds = Settings->HeartbeatTimeoutWithEditorSeconds;
```

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp [hunk 5]
```diff
@@ -101,17 +101,16 @@ struct ConfigureConnection
...
-		
+
 #else
 		TimeoutSeconds = Settings->HeartbeatTimeoutSeconds;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp:106]:
```suggestion
		float TimeoutSeconds = Settings->HeartbeatTimeoutSeconds;
```

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp (hunk 1 above)
Comment (razvan-at-improbable) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp:108]: Oh I see, I misunderstood the first time sorry

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp (hunk 3 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp:103]: Why resolve this? It looks like you've added a unnecessary new line.

**Comment by m-samiec** [comment]
Can you add a release note also please

File: CHANGELOG.md [hunk 6]
```diff
@@ -33,6 +33,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added a function that allows the worker coordinator to periodically restart the simulated player clients with a bunch of parameters. This feature is disabled by default and can be enabled via `max_lifetime` setting.
 - Exposing worker upstream/downstream window sizes as GDK options for both clients and servers, (`ClientDownstreamWindowSizeBytes`, `ClientUpstreamWindowSizeBytes`) and (`ServerDownstreamWindowSizeBytes` and `ServerUpstreamWindowSizeBytes`).
 - `bOnlyRelevantToOwner` is now supported. Ownership must be setup prior to the first replication of the Actor otherwise it will be ignored.
+- Added use of the heartbeat settings found in the editor for worker heartbeat parameters.
```
Comment (m-samiec) [inline CHANGELOG.md:36]:
```suggestion
- GDK heartbeat settings are now used to control the worker heartbeat configurations.
```

**Review by m-samiec — APPROVED** [review]
Nice!

**Pull request merged**